modules/volume/src/processors/volumeregionmap.cpp

```diff
@@ -0,0 +1,223 @@
+            dataFrameState = 1;
+        } else if (isSorted(src)) {  // Sorted + non continuous
+            dataFrameState = 2;
+        } else {
+            dataFrameState = 3;  // Unsorted + non continuous
+        }
+
+        size_t index = 0;
+        // std::map<unsigned int, unsigned int> orderedIndexMap;
+        std::unordered_map<unsigned int, unsigned int> unorderedIndexMap;
+        switch (dataFrameState) {
+            case 1:
+                std::transform(dataPtr, dataPtr + dim.x * dim.y * dim.z, dataPtr,
+                               [&](const ValueType& v) {
+                                   if (dst.size() - 1 < v) {
+                                       return static_cast<ValueType>(missingValue);
+                                   }
+                                   return static_cast<ValueType>(dst[static_cast<uint32_t>(v)]);
+                               });
+                break;
+            case 2:
+                std::transform(dataPtr, dataPtr + dim.x * dim.y * dim.z, dataPtr,
+                               [&](const ValueType& v) {
+                                   index = binarySearch(src, static_cast<uint32_t>(v));
+                                   return static_cast<ValueType>(dst[index]);
+                               });
+                break;
+            case 3:
+                for (size_t i = 0; i < src.size(); ++i) {
+                    unorderedIndexMap[src[i]] = dst[i];
+                }
+                std::transform(dataPtr, dataPtr + dim.x * dim.y * dim.z, dataPtr,
+                               [&](const ValueType& v) {
+                                   if (unorderedIndexMap.count(static_cast<uint32_t>(v)) == 1) {
+                                       return static_cast<ValueType>(
+                                           unorderedIndexMap[static_cast<uint32_t>(v)]);
+                                   }
+                                   return static_cast<ValueType>(missingValue);
+                               });
+                break;
+            default:
+                std::transform(dataPtr, dataPtr + dim.x * dim.y * dim.z, dataPtr,
+                               [&](const ValueType& v) {
+                                   for (size_t i = 0; i < src.size(); ++i) {
+                                       if (static_cast<uint32_t>(v) == src[i]) {
+                                           return static_cast<ValueType>(dst[i]);
+                                       }
+                                   }
+                                   return static_cast<ValueType>(missingValue);
+                               });
+                break;
+        }
+    });
+}
+
+/// <summary>
+/// Tests if a index vector contains sequential numbers without gaps.
+/// </summary>
+/// <param name="src">Index vecotr</param>
+/// <returns>True if sequential. False if not sequential.</returns>
+bool isSortedSequence(const std::vector<unsigned int>& src) {
```

Comment:
Equivalent to https://en.cppreference.com/w/cpp/algorithm/is_sorted ?

Reply:
This has now been changed.